reference/OpenAPI.json

```diff
@@ -32561,144 +32318,10 @@
-            "properties": {
-              "banner": {
-                "type": "string",
-                "description": "Name of the earning rule. This is displayed as a header for the earning rule in the Dashboard."
-              },
-              "object_id": {
-                "type": "string",
-                "description": "A unique campaign identifier assigned by the Voucherify API."
-              },
-              "object_type": {
-                "type": "string",
-                "description": "Defines the object associated with the earning rule. Defaults to `campaign`.",
-                "default": "campaign",
-                "enum": [
-                  "campaign"
-                ]
-              }
-            }
-          },
-          "object": {
-            "type": "string",
-            "default": "earning_rule",
-            "enum": [
-              "earning_rule"
-            ],
-            "description": "The type of the object represented by JSON. Default is earning_rule."
-          },
-          "automation_id": {
-            "type": "string",
-            "description": "For internal use by Voucherify."
-          },
-          "start_date": {
-            "type": "string",
-            "description": "Start date defines when the earning rule starts to be active. Activation timestamp is presented in the ISO 8601 format. Earning rule is inactive before this date. If you don't define the start date for an earning rule, it'll inherit the campaign start date by default."
-          },
-          "expiration_date": {
-            "type": "string",
-            "description": "Expiration date defines when the earning rule expires. Expiration timestamp is presented in the ISO 8601 format. Earning rule is inactive after this date.If you don't define the expiration date for an earning rule, it'll inherit the campaign expiration date by default."
-          },
-          "validity_timeframe": {
-            "$ref": "#/components/schemas/ValidityTimeframe"
-          },
-          "validity_day_of_week": {
-            "$ref": "#/components/schemas/ValidityDayOfWeek"
-          },
-          "validity_hours": {
-            "$ref": "#/components/schemas/ValidityHours"
-          },
-          "metadata": {
-            "type": "object",
-            "description": "The metadata object stores all custom attributes assigned to the earning rule. A set of key/value pairs that you can attach to an earning rule object. It can be useful for storing additional information about the earning rule in a structured format."
-          }
-        },
-        "required": [
-          "id",
-          "created_at",
-          "loyalty",
-          "source",
-          "object",
-          "automation_id",
-          "metadata"
-        ]
-      },
       "EarningRuleEvent": {
         "title": "Earning Rule Event",
-        "type": "string",
-        "enum": [
```

Comment:
Dlaczego jednak `enum` został usunięty?

Na tyle, na ile Cię zrozumiałem, to w takich przypadkach można bezpiecznie dawać `enum`, bo tylko ruch będzie dodaj/usuń do listy, ale nie edycja nazw zdarzeń.

Reply:
no właśnie :) ponieważ ta lista jest potencjalnie wrażliwa na zmiany, to tego z tego co sprawdzałem to można robić jakieś customowe earning rulke na bazie metadata, i przez to ta lista może mieć nieskończoną ilość wartości.

Reply:
Ok, czaję